### PR TITLE
Changes Mod Say Into an Admin-to-Mentor Message Verb

### PR DESCRIPTION
**Pull request by Krausus**
This is a repost of @LittleBigKid2000's #5378 ~~(plus a little fix), which was closed after what mostly amounted to "no, because we said so, and other important people said so, too, so *there!"* with a modest amount of *actual* criticism~~ -- initially, I also had a fix that actually gave the updated verb to mentors, but the headmins are confident that it would end up being abused. Now, `msay` is visible to mentors, but not normally usable by them.

In case admins need to have a discussion with the mentors, this also adds a new "Toggle Mentor Chat" verb in the "Server" tab, available to all admins. This will toggle and notify mentors of whether they can use `msay`. Note that it *is* disabled by default, and mentors will have no access to `msay` unless explicitly enabled by an admin.

:cl:
tweak: Admins can now use the 'msay' command to speak directly to all mentors.
rscadd: Added a new command, "Toggle Mentor Chat", to give mentors temporary access to 'msay'.
/:cl:

### CODE REVIEW

**Comment by LittleBigKid2000** [comment]
I tried already. I got ye-
>LittleBigKid didn't see fit to keep the PR open, but I still feel that this feature would be more beneficial than detrimental, so here we are again.
Oh.
Well, I closed it because I got an answer to the question of 'why not' I considered sufficient.

**Comment by Fox-McCloud** [comment]
This was universally rejected by all headmins, if I recall correctly, twice.

Mentors are not meant to be able to have any administrative capabilities what-so-ever. Whenever this has been tried, it always devolved into a buddy-group that talked about things they *shouldn't*.

**Comment by lordpidey** [comment]
I agree, there should be a way for admins to talk to all mentors simultaneously, but there is no need for a mentorchat.

**Comment by Krausus** [comment]
> This was universally rejected by all headmins, if I recall correctly, twice.

Then I'm sure they'll pop in and explain why they rejected it, and we can have a discussion about their reasoning.

> Mentors are not meant to be able to have any administrative capabilities what-so-ever.

Right, because talking to each other is administrative, but being able to directly message anyone (including each other) and seeing all such direct messages is totally different and totally fine.

> Whenever this has been tried, it always devolved into a buddy-group that talked about things they *shouldn't*.

If they're doing something they shouldn't, they should simply be told to knock it off. That logic is no better than removing anything else a player might abuse. If a single mentor is caught abusing direct messages, will mentors be removed to ensure no mentor can ever do so? No, because that's *absurd.*

**Comment by IcyV** [comment]
>Whenever this has been tried, it always devolved into a buddy-group that talked about things they shouldn't.

Very true and it's how it's always been. If people get called out on it, they just wait until they think nobody who can do something about it is on. It's just not a good system for people that low down on the chain.

**Comment by Fox-McCloud** [comment]
>Right, because talking to each other is administrative, but being able to directly message anyone (including each other) and seeing all such direct messages is totally different and totally fine.

This was never intended. There's a lot of things mentors have access to that they shouldn't, primarily because of checks for `client.holder` instead of holder rights.

What mentors were always *meant* to be is a group of people who could receive and answer mentorhelps, but they'd have literally zero other capabilities outside of this (aside from a colored name).

**Comment by Krausus** [comment]
> > Whenever this has been tried, it always devolved into a buddy-group that talked about things they shouldn't.
> 
> Very true and it's how it's always been.

How it's always been? When was the last time mentors could explicitly communicate like this? Mentors have had a dedicated permission level for over 1½ years, and I don't see any PRs since that gave them their own dedicated chat.

Surely you aren't assuming that our current mentors will behave the same as the mentors we had that long ago.

> > being able to directly message anyone (including each other) and seeing all such direct messages is totally different and totally fine.

> This was never intended. There's a lot of things mentors have access to that they shouldn't, primarily because checks for `client.holder` instead of holder rights.

Over 1½ years, and this was never "fixed"? It's almost as though it's not actually an issue whatsoever, and treating it like one is pointless.

**Comment by Fox-McCloud** [comment]
@Krausus 

Yeah, it was never fixed, fully. Originally mentors had wayyyy more permissions than they currently do, but that was corrected a good 6+ months after it was first implemented. 

Like I said, how mentors were added was absolutely awful and completely hacked together in a very short time frame without taking into consequence anything of when it was done.

**Comment by Dumbdumn5** [comment]
For reasons already stated in the past, I'm for this. I won't bring anything more than that up.

**Comment by IcyV** [comment]
Like @Fox-McCloud said, I was there when mentors were first a thing for Para. Luckily I was mostly managing admins at the time, but a specific admin or two have horror stories from all the work they had to do to make sure mentors and mods didn't abuse having a chat available to them. All I know is that, for a fact, people used it to buddy-up and borderline meta throughout a round. It was pretty much the biggest thing we had to keep an eye out for.

I am also very much assuming that current mentors will act similar to mentors of the past. It's not exactly my first run around the block of managing people, especially so in SS13. It's just a thing that happens. It's not even usually malicious. How terribly the system worked as a whole in the past just highlighted how bad it could be.

**Comment by Krausus** [comment]
> It was routinely complained about by admins and headmins /really/ disliked having to police not only the game's chat, but mentor chat as well.

> a specific admin or two have horror stories from all the work they had to do to make sure mentors and mods didn't abuse having a chat available to them.

This doesn't make sense. If specific mentors were problematic, why were they kept around? If nobody was actually problematic, then how was it such a problem?

It sounds more like a small number of admins were extremely strict about what they felt mentors should be allowed to discuss, but nobody else considered it enough of a problem to do anything about it, so they stuck to complaining without solving anything.

> I am also very much assuming that current mentors will act similar to mentors of the past. It's not exactly my first run around the block of managing people, especially so in SS13. It's just a thing that happens.

Wow, this sounds *really* cynical.

> It's not even usually malicious.

... if it can't be considered malicious, is it really a problem? Or was the problem that some admins blew things out of proportion?

> This is how they had their own chat to begin with because they inherited it from mod perms.

Ah, so they didn't *really* have their own chat, they just had access to the same chat the mods did, who presumably had more leeway about what they were allowed to discuss. *That* sounds like an annoying mix to try and monitor. But a mentor-specific chat isn't quite the same as that...

**Comment by Necaladun** [comment]
Not really for this - 

We have the discord for getting their attention and a few other things, or to talk shit about the hilarious bad questions, etc.

 I don't think it's best to add an in-game way to chat, for a lot of the reasons above, but mostly to keep mentors as...well, more connected to the players asking questions than each other. 

I don't see any real need for it, and struggle to think what it'd be used for.

**Comment by Earthdivine** [comment]
To be honest, I do not want this as it will just clutter up the admin chat and if the admins disable it due to clutter, they can't monitor the chat for metagaming or anything else.

However, if the admins will actually monitor and keep an eye on this chat so it can be used to confirm knowledge and not act like a chatroom, I am fine with it being in. They got the Discord Chat for the chatroom.

**Comment by Krausus** [comment]
> We have the discord for getting their attention

The one that, apparently, the mentors don't actually use? Are they even required to be in that chat?

> I don't think it's best to add an in-game way to chat

They can already chat through PMs, and do, as needed. It's not a great way to do so, however, and not logged separately from other PMs.

> I don't see any real need for it, and struggle to think what it'd be used for.

Some questions don't have an easy answer, and if a mentor isn't sure, their current options are responding "I don't know" or hoping another mentor responds, since they can't just ask if another mentor knows for sure. And if a mentorhelp asks or transitions to a question that only an admin can answer, the mentors are stuck either hoping an admin notices it, or bwoinking them directly.

For admins, I recall a recent admin event where the plan was to tell mentors to not answer specific questions... followed by an `msay`, then the realization that mentors don't hear those, and that there's no way to simply communicate with them. There's PMs, but those are meant for one-to-one communication.

> it will just clutter up the admin chat

The PMs mentors use to chat now do that already, but it doesn't seem to be an issue.

> if the admins disable it due to clutter, they can't monitor the chat for metagaming or anything else.

That won't be a problem, because there isn't a toggle for it.

> However, if the admins will actually monitor and keep an eye on this chat so it can be used to confirm knowledge and not act like a chatroom, I am fine with it being in.

This should be fine, then. Mentors *can* (and do) already communicate through PMs when they need to, and I've not heard of that being a problem, even though it's supposedly not intended.

**Comment by Fox-McCloud** [comment]
Going to be closing this. I consulted with the headmins on this, and they stated to go ahead and close this.

**Comment by Krausus** [comment]
> Going to be closing this. I consulted with the headmins on this, and they stated to go ahead and close this.

... what the *hell?* Two of the headmins have made a single comment, only one of them was explicitly opposed, and before either of them can comment again... it's closed?

You didn't even *attempt* to provide any justification for why it needed to be closed. If it turns out that mentors having a proper chat is unacceptable - which I'm not even sure about, seeing as you preempted the discussion with this closure - all I had to do is revert the commit that added the verb to mentors, and it becomes an admin-only mentor notification command.

This "close first, ask for compromise never" nonsense is toxic.

**Comment by CrAzYPiLoT-SS13** [comment]
Wow. I am just going to say, I didn't expect this level of distrust to our
current mentors and this amount of literal nonsense arguments. I have
resigned from my mentorship for personal reasons, but anyone who has ever
been a mentor, or even talked with out current mentors, know what level of
professionalism they uphold. Again, this PR and the issue of mentorchat has
not been dealt with professionally and I agree with Krausus on all points.

Once again, I was stunned by the level of distrust to the mentors, and it's
good to know how mentors are seen from their superiors' eyes: question
answering machines that can not have any trust at disposal to them, even
for a simple thing like mentorchat. I am sure that the mere thought of
turning a professional-atmosphered OOC channel wouldn't even cross mentors'
minds.

**Comment by PJB3005** [comment]
2¢ from somebody who doesn't even play on this server:

I see no reason to not have this, if mentors abuse it and it causes *any* then *maybe* you should remove the crap mentors.

**Comment by Necaladun** [comment]
This is not about distrust of mentors. We don't consider there to really be a risk of abuse.

There are a two main reasons behind this:
1) We don't want another chat to have to spam us admins - and if we disable it we can't monitor it
2) We don't want mentors to have too much of an elite clique feel, where they are disconnected from the playerbase.

Again, this is not about distrust of mentors at all.

**Comment by CrAzYPiLoT-SS13** [comment]
1) Why does it have to be mandatory for all admins to see? Make it a
preference.

2) Again, you say this is not about distrust of mentors, and yet you give
this reasoning. If you really trusted your mentors, you'd trust them with
their own OOC channel for the sake of simplicity of collaborating, not
because you wanted to give them a den to make a clique in. Both I and you,
and other headmins know that the little authority the mentors are given,
has in no way impacted their community relations. They are the same people
as before being given mentorship, they behave the same, and most
importantly, they don't distance themselves from the playerbase, let alone
form special snowflake cliques to hang out only with their mentor buddies.
I understand you say 'clique feel' and that's the entire point; even if you
gave mentors host permissions (stupid example, I know), I guarantee you,
they would not have 'an elite clique feel'.

**Comment by Necaladun** [comment]
1) If it's a prefernce and it's off, then we're not monitoring it.

2) Yes. It's not because we don't trust them. It's because we don't want them to be more distant or to be -seen- as being more distant. If we gave them host powers, they would be seen as being more distant and elite. Even if it was used 100% responsibly, the feel is important. Public perception is just as important as the reality.

Basically, there's no real need we see for this that isn't covered by the discord or the mentors themselves mentorhelping.

**Comment by CrAzYPiLoT-SS13** [comment]
See, this is the kind of reasoning you should have supplied in the first
place. Your post now provides some great points.

The community feel is a really important factor in all of this and I agree
with this point. Although, I do not feel the mentors will be seen as too
much of an elite group by the public, simply because of their constant
interaction. Take admins for example, they are supposed to be the, let's
say, law enforcement of the server and discord. Many people fear admins on
different servers and communities. In our community, that is not the case.
The admins constantly participate in every single thing community-wise, and
that's what makes them be seen as normal people and a regular part of the
community. Even while they have their own OOC channel. The same applies to
mentors. They have their own discord channel, and nothing has changed. They
still participate in conversations on the discord and on the server, they
mingle with people. Simply put, they are not seen and probably will not be
seen as an elite group that should be disregarded in normal circumstances.
Same goes for the admins.

As for the need for it, it is true there are two or three ways they can
communicate with each other, other than the mentorchat in question. These
ways of communication have their flaws, just like the mentorchat has its
own flaws. Take the discord channel, for example, it is mostly not active a
lot (or at least wasn't, while I was a mentor) and is almost never used for
mutual communication. Mentorhelps, on the other hand, are very often used,
but are impractical and simply not meant for mentor-to-mentor comms.

The mentorchat would solve these problems, but the monitoring thing remains.

**Comment by IcyV** [comment]
The cons of a mentorsay simply outweigh the potential single benefit of them being able to answer the occasional question that someone doesn't know. It would require it to be maintained and watched by an admin at pretty much all times to make sure it isn't abused. It would divert attention if anything goes on and it's a drain on other time and resources if an issue does pop up. It's a single pro and a list of cons.

Honestly, the problem lies in the mentors if they need a mentorsay to answer questions. If someone isn't being answered or a mentor doesn't know the answer, why isn't someone else stepping up and answering for them? If someone else could answer them in mentorsay, why couldn't said person have answered it themself?

**Comment by Krausus** [comment]
> This is not about distrust of mentors. We don't consider there to really be a risk of abuse.

> if we disable it we can't monitor it

> It's not because we don't trust them

So... you trust that mentors wouldn't abuse a mentor chat, but it *needs* to be monitored for abuse? It needs to be monitored for the abuse that you trust wouldn't happen. But monitoring it isn't feasible, because...

> We don't want another chat to have to spam us admins

... the admins will get spammed. They don't get spammed while monitoring the mentors that already can and do use PMs to talk to each other, but putting a different label on those would result in dramatically more use?

> We don't want mentors to have too much of an elite clique feel

> we don't want them to be more distant or to be -seen- as being more distant

They have a private Discord chat to use out-of-game (is *that* constantly monitored?), and PMs to use in-game. They already have all the means they need to huddle up and clique. But it doesn't happen, does it?

And if we wanted them to interact more directly with the playerbase, we'd have them answering questions in OOC.

> It would require it to be maintained and watched by an admin at pretty much all times to make sure it isn't abused. It would divert attention if anything goes on and it's a drain on other time and resources if an issue does pop up. It's a single pro and a list of cons.

Geez. You could say this about *literally anything* that has the slightest potential for abuse. And though this would be a great argument against mid-round OOC (maybe we should disable that to reduce admin workload?), you're talking about trusted mentors, who had to submit an application and get approval from multiple admins. We don't just pluck unruly children off the street to work in the answer factory.

**Comment by Krausus** [comment]
Well, I give up. I wasn't anticipating the intense "not-distrust" even the headmins have for our mentors, and I can't really argue against it if it can't be rationalized.

I've reverted the fix commit, and updated the PR description to explain what that means:

> Now, it simply changes `msay` so mentors also receive the message.

> This **does not** give mentors access to the `msay` verb. Technically, it's now coded such that, if an admin manually gives a mentor the verb, they can use it; however, there's no way for them to get the verb and use mentor chat otherwise, so it is effectively an admin-only method of directly speaking to all mentors.

I assume nobody has a problem with turning the unused `msay` into a way for admins to speak to all connected mentors, so this is probably good to be reopened, @Fox-McCloud.

**Comment by FlattestGuitar** [comment]
I've had to myself tell memetors to leave some poor guy alone, because they didn't have the right information to help him and were only making things worse. 

I used mentorhelp for that and I don't quite see why that won't work in the future.

**Comment by Earthdivine** [comment]
I would rather just use a mentorwide msay then use Mentorhelp for that. Sure you get a powink from using the help command but any mentors who need something told, like "Hey heads up this was changed", is a bit more noticeable with msay considering it's not something that appears on the chat log often, compared to MentorHelp.

**Comment by Krausus** [comment]
In addition to blending in with normal mentorhelps, contacting mentors like that is a misuse of the "ask mentors a question" system, bwoinks normal admins, and gets logged (to the server log *and* Discord) as a mentorhelp question.

**Comment by Fox-McCloud** [comment]
re-opening, for now; will await headmin input.

**Comment by Aurorablade** [comment]
honestly..the mentorhelp thing seems like it would work better. but that jue me...

**Comment by Aurorablade** [comment]
though re reading it..it MIGHT be good...i am not sure...

**Comment by HereticChurch** [comment]
Why not make it something that starts off disabled every round, but give the admins the ability to turn it on for all mentors, giving them all the ability to use it. Would let admins get some use out of it and chat with all mentors at once without using PMs or let mentors chat about answers and whatnot, and then afterwords can be disabled again. No real harm in that, I feel.

**Comment by Krausus** [comment]
Good idea. Updated:

> In case admins need to have a discussion with the mentors, this also adds a new "Toggle Mentor Chat" verb in the "Server" tab, available to all admins. This will toggle and notify mentors of whether they can use `msay`. Note that it *is* disabled by default, and mentors will have no access to `msay` unless explicitly enabled by an admin.
